src/opensearchDatasource.ts

```diff
@@ -1114,9 +1114,10 @@ export class OpenSearchDatasource
 
       const buckets = res.responses[0].aggregations['1'].buckets;
       return _.map(buckets, (bucket) => {
+        const keyString = String(bucket.key);
         return {
-          text: bucket.key_as_string || bucket.key,
-          value: bucket.key,
+          text: bucket.key_as_string || keyString,
+          value: isTagValueQuery ? keyString : bucket.key,
```

Comment:
Technically `MetricFindValue` is supposed to accept numbers as values, but it was logging a console error instead of adding the filter when I tried it, it worked properly when it was a string.